src/Gas.cpp

```diff
@@ -0,0 +1,167 @@
+#include "Gas.h"
+
+/*
+Stuff for primitive state
+*/
+
+IdealGas::PrimitiveState::PrimitiveState() :
+  // initialiser list
+  rho{0},
+  u{0,0},
+  p{0} 
+  // empty body...
+  {};
+  
+
+Precision IdealGas::PrimitiveState::getSoundSpeed()
+{
+  return std::sqrt( GAMMA * getP() / getRho() );
+}
+
+
+Precision IdealGas::PrimitiveState::getEnergy()
+{
+  return 0.5 * getRho() * getUSquared() + getP() / GM1;
+}
+
+/* getters and setters for PrimitiveState  */
+
+void      IdealGas::PrimitiveState::setRho(const Precision val){
+  rho = val;
+}
+
+Precision IdealGas::PrimitiveState::getRho() const{
+  return rho;
+}
+
+void      IdealGas::PrimitiveState::setU(int index, const Precision val){
+  // bounds checking? why not.... if we have time
```

Comment:
what bounds are you checking here?

Reply:
was checking that we didn't pass invalid index. should be fine to remove